test/tools/yulopti.cpp

```diff
@@ -214,47 +359,48 @@ class YulOpti
 					case '#':
 						return;
 					case ',':
-						VarNameCleaner::run(m_context, *m_ast);
+						runVarNameCleaner();
 						// VarNameCleaner destroys the unique names guarantee of the disambiguator.
 						disambiguated = false;
 						break;
 					case ';':
-					{
-						Object obj;
-						obj.code = m_ast;
-						StackCompressor::run(m_dialect, obj, true, 16);
+						runStackCompressor();
 						break;
-					}
 					default:
-						OptimiserSuite{m_context}.runSequence(
-							std::string_view(&option, 1),
-							*m_ast
-						);
+						runSequence(string_view(&option, 1));
 				}
-				_source = AsmPrinter{m_dialect}(*m_ast);
+
+				resetNameDispenser();
+				runCodeAnalyzer(errorReporter);
 			}
 			catch (...)
 			{
 				cerr << endl << "Exception during optimiser step:" << endl;
 				cerr << boost::current_exception_diagnostic_information() << endl;
 			}
 			cout << "----------------------" << endl;
-			cout << _source << endl;
+			printObject();
 		}
 	}
 
 private:
-	shared_ptr<yul::Block> m_ast;
+	shared_ptr<yul::Object> m_object;
+	bool m_inputWasCodeBlock = false;
```

Comment:
Isn't it still a code block? A block should not be converted to an object by the tool.
```suggestion
	bool m_inputIsCodeBlock = false;
```

Reply:
Renamed.

The tool now uses `ObjectParser::parse` which has a special case for code blocks. That makes it possible to handle objects and code blocks the same for the most part. Are there issues with that approach?

Reply:
It's fine as long as you don't print an object when the input was a block.